### PR TITLE
[JENKINS-69658] - failed-test.jelly javascript un-inlined.

### PR DESCRIPTION
**Pull request by Artmorse**
**The issue**
You can find all the issue details [here](https://issues.jenkins.io/browse/JENKINS-69658).

**My updates**
I've moved the `onclick` calls from the `failed-test.jelly` file to the `failureSummary.js` one.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/lib/hudson/test/js/failureSummary.js [hunk 1]
```diff
@@ -18,3 +18,24 @@ function hideFailureSummary(id) {
...
 }
+
+
+document.addEventListener('DOMContentLoaded', (event) => {
+
+    const testShowlinks = document.querySelectorAll("a[id*=-showlink]");
+    testShowlinks.forEach((element) => {
+        element.onclick = (_) => {
+            const testId = element.id.replace('-showlink', '');
+            showFailureSummary(testId, document.URL + "/summary");
```
Comment (yaroslavafenkin) [inline src/main/resources/lib/hudson/test/js/failureSummary.js:29]: `document.URL` is wrong, as it refers to that page we're in. Request has to go to `JENKINS_URL/job/fs/5/testReport/<package_name>/<test_name>/summary` and not just `JENKINS_URL/job/fs/5/testReport/summary`
Currently this returns 404:
image

**Review by yaroslavafenkin — CHANGES_REQUESTED** [review]

File: src/main/resources/lib/hudson/test/js/failureSummary.js (hunk 1 above)
Comment (Artmorse) [reply src/main/resources/lib/hudson/test/js/failureSummary.js:29]: Ok, can you please complete the **Reproduction steps** section of the [Jira ticket (JENKINS-69658)](https://issues.jenkins.io/browse/JENKINS-69658) description ?

Comment (Artmorse) [reply src/main/resources/lib/hudson/test/js/failureSummary.js:29]: ping @yaroslavafenkin

Comment (Artmorse) [reply src/main/resources/lib/hudson/test/js/failureSummary.js]: Perfect! :)

**Comment by Artmorse** [comment]
@yaroslavafenkin 
Can you add the `hacktoberfest-accepted` label to this PR if it's seems good to you ? :blush:

**Review by yaroslavafenkin — COMMENTED** [review]
Still does not work as expected.

image

Request URL still looks wrong: 
image

Providing steps I used:
1. Create a freestyle project
2. Select Git as SCM, as repository URL specify `https://github.com/yaroslavafenkin/security-notifier.git`. As branch specifier put `*/fail`.
3. Add 'Invoke top-level Maven targets' build step. As goal set `test`.
4. Add 'Publish JUnit test result report' post-build action. As fileset includes put `**/*.xml`.
5. Save it and trigger a build.
6. When it fails go to the build and then to 'Test Result' action.
7. Open browser developer tools and go under 'All Failed Tests' hit the plus icon to obtain details about the failed test. You'll see failed request in the console, and should see an error as on the screenshot above.

**Comment by Kevin-CB** [comment]
Hey @Artmorse, I tried reviewing this PR and followed Yaroslav's steps. I'm not encountering any errors, it's working as expected.

However, I noticed that we missed to report an occurrence of inline JS ([summary.jelly#L34-L35](https://github.com/Artmorse/junit-plugin/blob/909f4d9978e8e161717ab46bd65cc43c3f3adc90/src/main/resources/hudson/tasks/junit/CaseResult/summary.jelly#L34-L35)) 
It's in a different class, but closely related to your changes, so if you're still willing to work on this topic, I'd be here to review it once more and help get it finally merged!

**Comment by Artmorse** [comment]
hey @Kevin-CB, yes I'll make the changes and I'll ping you when it's done!

**Pull request merged**